### ISSUE TITLE
flatten lockfile

### ISSUE THREAD

**Issue by XhmikosR** [dependabot/feedback#776]
So, this situation has bit me numerous times :)

Basically, after a while using dependabot, one ends up with many duplicate packages. A recent example is https://github.com/twbs/bootstrap/pull/29730

I'm not sure how dependabot could mitigate this issue. `npm dedupe` does not seem to solve the issue. On the other hand, regenerating the lock file shouldn't be done by default since it can break many things.

I know there is https://github.com/dependabot/feedback/issues/313, but it seems that was focused mostly on yarn.

**Comment by feelepxyz** [comment]
@XhmikosR Thanks for submitting! I've attempted to fix this in the past but didn't even get the npm dedupe command working properly as it sometimes ended up breaking the lockfile. I think this should be solved in npm itself because it probably requires some internal changes.

We're super stretched at the moment so don't have much spare capacity to work on wider improvements for the next few months.

**Comment by XhmikosR** [comment]
@feelepxyz NP, I understand. :)

Like I said, I don't even know how you could fix it on your side either. I will try to raise an issue with npm

**Comment by XhmikosR** [comment]
@feelepxyz which command does dependabot run? `npm i package@version`?

**Comment by feelepxyz** [comment]
@XhmikosR yes pretty much but Dependabot doesn't call the CLI directly, instead it uses the library functions to avoid downloading `node_modules` and calling life-cycle hooks: https://github.com/dependabot/dependabot-core/blob/master/npm_and_yarn/helpers/lib/npm/updater.js#L20

**Comment by XhmikosR** [comment]
I wonder if the same issue happens when running `npm i package@version`...

**Comment by XhmikosR** [comment]
This is starting to be very annoying :/

https://github.com/twbs/bootstrap/pull/30951

I have to redo the lock file every couple of weeks.

**Comment by XhmikosR** [comment]
This is starting to be very annoying :/

https://github.com/twbs/bootstrap/pull/30951/files

I have to redo the lock file every couple of weeks.
